### ISSUE TITLE
package ‘argosTrack’ is not available (for R version 3.3.2)

### ISSUE THREAD

**Issue by autumnlynn** [calbertsen/argosTrack#1]
Hello --

I'm writing to inquire if this package is still being maintained (and plans for long-term maintenance). I received an error upon install that it is not available for 3.3 (and R is now on 3.4). Seems to have some advantages over crawl and was hoping to try it out. 

Thanks!

**Comment by calbertsen** [comment]
Hello,

The package is being maintained, however, it is not yet on CRAN, so you have to install it using the `devtools` package.

On Windows, you need to install Rtools from this page: https://cran.r-project.org/bin/windows/Rtools/
Then you can install `devtools`, `TMB`, and `RcppEigen` from CRAN:
```r
install.packages(c("devtools","RcppEigen","TMB"))
```
Finally, the `argosTrack` package can be installed by running:
```r
library(devtools)
install_github("calbertsen/argosTrack")
```

Let me know if there is any trouble installing this way.

Christoffer

**Comment by autumnlynn** [comment]
Hi Christoffer -

Thanks very much for the information and the reply.

I did previously attempt installing using devtools, however, compilation is
failing and I’m being tossed multiple errors. Having a bit of trouble
discerning whether it’s on my end, or the package compatibility, etc. TMB
seems to be installed.

If you’ve seen any of the below errors previously, I’d appreciate the help!

Cheers,
Autumn-Lynn

Installing TMB
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file
--no-environ --no-save --no-restore  \
  --quiet CMD INSTALL  \

'/private/var/folders/_3/gjxy29_n0m783b3wxlkpychr0000gn/T/RtmpMm3MHN/devtoolse43e750b3c86/TMB'
 \

--library='/Library/Frameworks/R.framework/Versions/3.3/Resources/library'
--install-tests

* installing *binary* package ‘TMB’ ...
* DONE (TMB)
'/Library/Frameworks/R.framework/Resources/bin/R' --no-site-file
--no-environ --no-save --no-restore  \
  --quiet CMD INSTALL  \

'/private/var/folders/_3/gjxy29_n0m783b3wxlkpychr0000gn/T/RtmpMm3MHN/devtoolse43e402be758/calbertsen-argosTrack-5c38754'
 \

--library='/Library/Frameworks/R.framework/Versions/3.3/Resources/library'
--install-tests

* installing *source* package ‘argosTrack’ ...
** libs
clang++ -I/Library/Frameworks/R.framework/Resources/include -DNDEBUG
 -I/usr/local/include -I/usr/local/include/freetype2 -I/opt/X11/include
-I"/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include"
-I"/Library/Frameworks/R.framework/Versions/3.3/Resources/library/RcppEigen/include"
  -fPIC  -Wall -mtune=core2 -g -O2  -c init.cpp -o init.o
In file included from init.cpp:7:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/TMB.hpp:36:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/Rstream.hpp:10:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:38:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:216:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:871:34:
error: too many arguments provided to function-like macro invocation
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
                                 ^
/Library/Frameworks/R.framework/Resources/include/Rinternals.h:1236:9:
note: macro 'length' defined here
#define length(x)               Rf_length(x)
        ^
In file included from init.cpp:7:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/TMB.hpp:36:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/Rstream.hpp:10:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:38:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:216:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:871:15:
error: expected ';' at end of declaration list
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:956:34:
error: too many arguments provided to function-like macro invocation
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
                                 ^
/Library/Frameworks/R.framework/Resources/include/Rinternals.h:1236:9:
note: macro 'length' defined here
#define length(x)               Rf_length(x)
        ^
In file included from init.cpp:7:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/TMB.hpp:36:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/Rstream.hpp:10:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:38:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:216:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:956:15:
error: expected ';' at end of declaration list
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:1040:34:
error: too many arguments provided to function-like macro invocation
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
                                 ^
/Library/Frameworks/R.framework/Resources/include/Rinternals.h:1236:9:
note: macro 'length' defined here
#define length(x)               Rf_length(x)
        ^
In file included from init.cpp:7:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/TMB.hpp:36:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/Rstream.hpp:10:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:38:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:216:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:1040:15:
error: expected ';' at end of declaration list
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
              ^
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:1126:34:
error: too many arguments provided to function-like macro invocation
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
                                 ^
/Library/Frameworks/R.framework/Resources/include/Rinternals.h:1236:9:
note: macro 'length' defined here
#define length(x)               Rf_length(x)
        ^
In file included from init.cpp:7:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/TMB.hpp:36:
In file included from
/Library/Frameworks/R.framework/Versions/3.3/Resources/library/TMB/include/Rstream.hpp:10:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/iostream:38:
In file included from
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/ios:216:
/Library/Developer/CommandLineTools/usr/bin/../include/c++/v1/__locale:1126:15:
error: expected ';' at end of declaration list
    int length(state_type& __st, const extern_type* __frm, const
extern_type* __end, size_t __mx) const
              ^
8 errors generated.
make: *** [init.o] Error 1
ERROR: compilation failed for package ‘argosTrack’
* removing
‘/Library/Frameworks/R.framework/Versions/3.3/Resources/library/argosTrack’
Installation failed: Command failed (1)


From: Christoffer Moesgaard Albertsen <notifications@github.com>
<notifications@github.com>
Reply: calbertsen/argosTrack
<reply@reply.github.com>
<reply@reply.github.com>
Date: March 27, 2018 at 2:25:03 PM
To: calbertsen/argosTrack <argostrack@noreply.github.com>
<argostrack@noreply.github.com>
Cc: autumnlynn <autumnlynn.harrison@gmail.com>
<autumnlynn.harrison@gmail.com>, Author <author@noreply.github.com>
<author@noreply.github.com>
Subject:  Re: [calbertsen/argosTrack] package ‘argosTrack’ is not available
(for R version 3.3.2) (#1)

Hello,

The package is being maintained, however, it is not yet on CRAN, so you
have to install it using the devtools package.

On Windows, you need to install Rtools from this page:
https://cran.r-project.org/bin/windows/Rtools/
Then you can install devtools, TMB, and RcppEigen from CRAN:

install.packages(c("devtools","RcppEigen","TMB"))

Finally, the argosTrack package can be installed by running:

library(devtools)
install_github("calbertsen/argosTrack")

Let me know if there is any trouble installing this way.

Christoffer

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/calbertsen/argosTrack/issues/1#issuecomment-376626634>,
or mute the thread

**Comment by calbertsen** [comment]
Hi Autumn-Lynn,

Does it work if you install using:
```r
install_github("calbertsen/argosTrack",ref="mac")
```
?

Best,
Christoffer

**Comment by autumnlynn** [comment]
Success :) Thank you. Looking forward to exploring. Sorry to bother you if
I missed that somewhere!



From: Christoffer Moesgaard Albertsen <notifications@github.com>
<notifications@github.com>
Reply: calbertsen/argosTrack
<reply@reply.github.com>
<reply@reply.github.com>
Date: March 27, 2018 at 3:32:07 PM
To: calbertsen/argosTrack <argostrack@noreply.github.com>
<argostrack@noreply.github.com>
Cc: autumnlynn <autumnlynn.harrison@gmail.com>
<autumnlynn.harrison@gmail.com>, Author <author@noreply.github.com>
<author@noreply.github.com>
Subject:  Re: [calbertsen/argosTrack] package ‘argosTrack’ is not available
(for R version 3.3.2) (#1)

Hi Autumn-Lynn,

Does it work if you install using:

install_github("calbertsen/argosTrack",ref="mac")

?

Best,
Christoffer

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/calbertsen/argosTrack/issues/1#issuecomment-376647079>,
or mute the thread

**Comment by calbertsen** [comment]
The fix is merged to the master branch in 602fe8f0169e7a0392e8eeac09ef3bfb8d5289e6
